src/systems/performer_detector/PerformerDetector.hh

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright (C) 2020 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#ifndef IGNITION_GAZEBO_SYSTEMS_PERFORMERDETECTOR_HH_
+#define IGNITION_GAZEBO_SYSTEMS_PERFORMERDETECTOR_HH_
+
+#include <memory>
+#include <string>
+#include <unordered_set>
+
+#include <ignition/transport/Node.hh>
+
+#include "ignition/gazebo/Model.hh"
+#include "ignition/gazebo/System.hh"
+
+namespace ignition
+{
+namespace gazebo
+{
+// Inline bracket to help doxygen filtering.
+inline namespace IGNITION_GAZEBO_VERSION_NAMESPACE {
+namespace systems
+{
+
+  class IGNITION_GAZEBO_VISIBLE PerformerDetector
```

Comment:
Mind adding some documentation? :innocent:  Important to mention whether the detection happens on intersection or full containment, and what the published message means.

Reply:
https://github.com/ignitionrobotics/ign-gazebo/pull/125/commits/5d4efb6bdd8cc7f4de86d99a52e3d5f0b6a83e93